src/titiler/core/titiler/core/utils.py

```diff
@@ -116,3 +119,52 @@ def render_image(
         ),
         output_format.mediatype,
     )
+
+
+T = TypeVar("T")
+
+ValidParams = Dict[str, Any]
+Errors = List[Any]
+
+
+def get_dependency_query_params(
+    dependency: Callable,
+    params: Dict,
+) -> Tuple[ValidParams, Errors]:
+    """Check QueryParams for Query dependency.
+
+    1. `get_dependant` is used to get the query-parameters required by the `callable`
+    2. we use `request_params_to_args` to construct arguments needed to call the `callable`
+    3. we call the `callable` and catch any errors
+
+    Important: We assume the `callable` in not a co-routine.
+    """
+    dep = get_dependant(path="", call=dependency)
+    return request_params_to_args(
+        dep.query_params, QueryParams(urlencode(params, doseq=True))
```

Comment:
Notice the addition of `QueryParams` here. 
The reason is that `request_params_to_args` expects to see the result of this function. 
An example that helped me catch it is `rescale` param - [link to def](https://github.com/developmentseed/titiler/blob/main/src/titiler/core/titiler/core/dependencies.py#L441-L448).
```
rescale: Annotated[
    Optional[List[str]],
    Query(...)
]
```
If I pass things directly from STAC `renders` block, it would not pass validation, since it is defined as `[float]`
To mitigate this, I first serialize params to the state that fastapi would receive them, and then deserialize how it would do it.